src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -214,9 +214,9 @@ public FolderFileStorage(Builder<FolderFileStorageOptionsBuilder, FolderFileStor
             
         }
 
-        public Task<IEnumerable<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null, CancellationToken cancellationToken = default) {
+        public async Task<FileListResult> GetFileListAsync(string searchPattern = null, int? limit = null, CancellationToken cancellationToken = default) {
```

Comment:
Shouldn't this return an interface?

Reply:
FileSpec isn’t and I ran into issues with trying to make it return an interface.